packages/swr/src/index.ts

```diff
@@ -235,6 +236,12 @@ const generateSwrImplementation = ({
   doc?: string;
 }) => {
   const swrProps = toObjectString(props, 'implementation');
+
+  const hasParamNameQuery = props.some(
```

Comment:
OK then rename `hasParamNamedQuery` to `hasParamReservedWord` so people know why you are looking specifically for the word `query` they will know its a reserved word.

Reply:
@melloware 

Please allow me to add to the explanation.

The purpose of using `hasParamNameQuery` in the variable name is not whether it is a reserved word or not, but simply to check whether `query` is used as a parameter.

Let's consider a case where another reserved word has increased.
For example, suppose you want to treat `swrFn` as a reserved word and add `_` to it.
In that case, I think the following will be added:

```typescript
const hasParamNameSwrFn = props.some(
  (param: GetterProp) => param.name === 'swrFn',
);
const queryResultVarName = hasParamNameSwrFn ? '_swrFn' : 'swrFn';
```

Because, variables used within functions with the same name are marked with `_` depending on the existence of individual specified parameters, not whether the parameters contain reserved words. However, please note that this is just an example and there are other options if you actually want to respond.

From this, I think that the original `hasParamNameQuery` represents the original intention of the variable name rather than `hasParamReservedWord` as commented.

And I thought that the reason was expressed in the code, so it was sufficient, but what do you think?

Reply:
I disagree the word `query` is to generic in terms of React Query, Vue Query, SWR?   So this code tells me no reason wht you are looking for `hasParamNamedQuery` its too specific with no reasoning that what you are doing is.

"looking for a reserved word, preventing its use by appending an underscore so it does not collide with other usages"  From looking at your code I can't tell any of that and no where does it mention `query` is a reserved word you are looking for.

So someone looking at this code in 1 year will have the spend the time to think "why are they doing this" where as naming things properly like "checkReservedWord" or "cleanReservedWords" tells the developer exactly what you are doing without thinking about it.

Reply:
At the very least put a comment above the line.

```js
// #800 Convert param named query to _query to prevent collision
```

Reply:
@melloware 
Got it. I respect your opinion and change the function name from "hasParamNamedQuery" to "hasParamReservedWord".

Reply:
Fixed as bellow:

https://github.com/anymaniax/orval/compare/5693ebe1387492d5c25e8bc0c512fd4e8db63148..146f9bb90da54fbb012f2948c65280a445ca70bf